cmd/generate.go

```diff
@@ -0,0 +1,294 @@
+}
+
+// generateCmd represents the generate command
+var generateCmd = &cobra.Command{
+	Use:     "generate",
+	Aliases: []string{"gen"},
+	Short:   "generate a Clos topology file, based on provided flags",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if name == "" {
+			return errors.New("provide a lab name with --name flag")
+		}
+		licenses, err := parseFlag(kind, license)
+		if err != nil {
+			return err
+		}
+		log.Debugf("parsed licenses: %+v", licenses)
+
+		images, err := parseFlag(kind, image)
+		if err != nil {
+			return err
+		}
+		log.Debugf("parsed images: %+v", images)
+
+		nodeDefs, err := parseNodesFlag(kind, nodes...)
+		if err != nil {
+			return err
+		}
+		log.Debugf("parsed nodes definitions: %+v", nodeDefs)
+
+		b, err := generateTopologyConfig(name, mgmtNetName, mgmtIPv4Subnet.String(), mgmtIPv6Subnet.String(), images, licenses, nodeDefs...)
+		if err != nil {
+			return err
+		}
+		log.Debugf("generated topo: %s", string(b))
+		if file != "" {
+			err = saveTopoFile(file, b)
+			if err != nil {
+				return err
+			}
+		}
+		if deploy {
+			reconfigure = true
+			if file == "" {
+				file = fmt.Sprintf("%s.yaml", name)
+				err = saveTopoFile(file, b)
+				if err != nil {
+					return err
+				}
+			}
+			topo = file
+			return deployCmd.RunE(deployCmd, nil)
+		}
+		if file == "" {
+			fmt.Println(string(b))
+		}
+		return nil
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(generateCmd)
+	generateCmd.Flags().StringVarP(&mgmtNetName, "network", "", "", "management network name")
+	generateCmd.Flags().IPNetVarP(&mgmtIPv4Subnet, "ipv4-subnet", "4", net.IPNet{}, "management network IPv4 subnet range")
+	generateCmd.Flags().IPNetVarP(&mgmtIPv6Subnet, "ipv6-subnet", "6", net.IPNet{}, "management network IPv6 subnet range")
+	generateCmd.Flags().StringSliceVarP(&image, "image", "", []string{}, "container image name, can be prefixed with the node kind. <kind>=<image_name>")
+	generateCmd.Flags().StringVarP(&kind, "kind", "", "srl", fmt.Sprintf("container kind, one of %v", supportedKinds))
+	generateCmd.Flags().StringSliceVarP(&nodes, "nodes", "", []string{}, "comma separated nodes definitions in format <num_nodes>:<kind>:<type>, each defining a Clos network stage")
+	generateCmd.Flags().StringSliceVarP(&license, "license", "", []string{}, "path to license file, can be prefix with the node kind. <kind>=/path/to/file")
+	generateCmd.Flags().StringVarP(&nodePrefix, "node-prefix", "", defaultNodePrefix, "prefix used in node names")
+	generateCmd.Flags().StringVarP(&groupPrefix, "group-prefix", "", defaultGroupPrefix, "prefix used in group names")
+	generateCmd.Flags().StringVarP(&file, "file", "", "", "file path to save generated topology")
+	generateCmd.Flags().BoolVarP(&deploy, "deploy", "", false, "deploy a fabric based on the generated topology file")
+}
+
+func generateTopologyConfig(name, network, ipv4range, ipv6range string, images map[string]string, licenses map[string]string, nodes ...nodesDef) ([]byte, error) {
```

Comment:
might be opinionated:

I would have renamed the `nodes ...nodesDef` to `tiers ...tiersDef` to be inline with our logic of CLOS fabric to be composed out of tiers.

Reply:
let's keep it like this, who knows, we might add other topologies in the future...